swap/cashout.go

```diff
@@ -30,54 +31,104 @@ import (
 // CashChequeBeneficiaryTransactionCost is the expected gas cost of a CashChequeBeneficiary transaction
 const CashChequeBeneficiaryTransactionCost = 50000
 
-// CashoutProcessor holds all relevant fields needed for processing cashouts
-type CashoutProcessor struct {
-	backend    chain.Backend     // ethereum backend to use
-	privateKey *ecdsa.PrivateKey // private key to use
-}
+// CashoutRequestHandlerID is the handlerID used by the CashoutProcessor for CashoutRequests
+const CashoutRequestHandlerID = "CashoutProcessor_CashoutRequest"
 
 // CashoutRequest represents a request for a cashout operation
 type CashoutRequest struct {
 	Cheque      Cheque         // cheque to be cashed
 	Destination common.Address // destination for the payout
 }
 
-// ActiveCashout stores the necessary information for a cashout in progess
-type ActiveCashout struct {
-	Request         CashoutRequest // the request that caused this cashout
-	TransactionHash common.Hash    // the hash of the current transaction for this request
+// CashoutProcessor holds all relevant fields needed for processing cashouts
+type CashoutProcessor struct {
+	backend              chain.Backend     // ethereum backend to use
+	txScheduler          chain.TxScheduler // transaction queue to use
+	cashoutResultHandler CashoutResultHandler
+	cashoutDone          chan *CashoutRequest
+}
+
+// CashoutResultHandler is an interface which accepts CashChequeResults from a CashoutProcessor
+type CashoutResultHandler interface {
+	// Called by the CashoutProcessor when a CashoutRequest was successfully executed
+	// It will be called again if an error is returned
+	HandleCashoutResult(request *CashoutRequest, result *contract.CashChequeResult, receipt *types.Receipt) error
 }
 
 // newCashoutProcessor creates a new instance of CashoutProcessor
-func newCashoutProcessor(backend chain.Backend, privateKey *ecdsa.PrivateKey) *CashoutProcessor {
-	return &CashoutProcessor{
-		backend:    backend,
-		privateKey: privateKey,
+func newCashoutProcessor(txScheduler chain.TxScheduler, backend chain.Backend, privateKey *ecdsa.PrivateKey, cashoutResultHandler CashoutResultHandler) *CashoutProcessor {
+	c := &CashoutProcessor{
+		backend:              backend,
+		txScheduler:          txScheduler,
+		cashoutResultHandler: cashoutResultHandler,
 	}
-}
 
-// cashCheque tries to cash the cheque specified in the request
-// after the transaction is sent it waits on its success
-func (c *CashoutProcessor) cashCheque(ctx context.Context, request *CashoutRequest) error {
-	cheque := request.Cheque
-	opts := bind.NewKeyedTransactor(c.privateKey)
-	opts.Context = ctx
+	txScheduler.SetHandlers(CashoutRequestHandlerID, &chain.TxRequestHandlers{
```

Comment:
They are all used in tests at least. Technically all places which schedule transactions also should monitor at least `NotifyCancelled` and `NotifyStatusUnkown`. Those handlers will be added soon, once we have decided what we actually want to do in those scenarios (e.g. for cashing we might want to retry on `Cancelled` and log on `StatusUnknown`). Anyway I decided to leave this out to keep the PR smaller.

Anyway those handlers are important already as otherwise we cannot properly test the error conditions during the txqueue tests.

Reply:
Ok. Maybe, for now, we can already register the handlers for now and just write a log output?

Reply:
done